tests/amy/test_amy.py

```diff
@@ -374,3 +398,1223 @@ def _generate_day_from_index(i: int) -> str:
+
+    @pytest.mark.advanced
+    @patch("climakitae.explore.amy.get_data")
+    def test_retrieve_profile_data_no_delta_mode(self, mock_get_data):
+        """Test no_delta mode only returns future data."""
+        mock_future_data = MagicMock(spec=xr.Dataset)
+        mock_get_data.return_value = mock_future_data
+
+        kwargs_no_delta = {**self.valid_kwargs, "no_delta": True}
+        historic_data, future_data = retrieve_profile_data(**kwargs_no_delta)
+
+        assert historic_data is None
+        assert future_data == mock_future_data
+        assert mock_get_data.call_count == 1
+
+    @pytest.mark.advanced
+    @patch("climakitae.explore.amy.get_data")
+    def test_retrieve_profile_data_parameters_passed_correctly(self, mock_get_data):
+        """Test that parameters are passed correctly to get_data."""
+        mock_data = MagicMock(spec=xr.Dataset)
+        mock_get_data.return_value = mock_data
+
+        test_kwargs = {
+            "variable": "Precipitation",
+            "resolution": "9 km",
+            "warming_level": [3.0],
+            "cached_area": "Los Angeles County",
+            "units": "mm/day",
+            "latitude": 34.0,
+            "longitude": -118.0,
+        }
+
+        retrieve_profile_data(**test_kwargs)
+
+        # Check first call (historic data)
+        historic_call = mock_get_data.call_args_list[0][1]
+        assert historic_call["variable"] == "Precipitation"
+        assert historic_call["resolution"] == "9 km"
+        assert historic_call["warming_level"] == [1.2]  # Always 1.2 for historic
+        assert historic_call["downscaling_method"] == "Dynamical"
+        assert historic_call["timescale"] == "hourly"
+        assert historic_call["area_average"] == "Yes"
+        assert historic_call["approach"] == "Warming Level"
+
+        # Check second call (future data)
+        future_call = mock_get_data.call_args_list[1][1]
+        assert future_call["variable"] == "Precipitation"
+        assert future_call["warming_level"] == [3.0]  # User-specified warming level
+
+
+class TestComputeProfile:
+    """Test class for compute_profile function."""
+
+    def setup_method(self):
+        """Set up test fixtures for each test method."""
+        pass
+
+    def create_mock_profile_data(
+        self, warming_levels=None, simulations=None, days_in_year=365
+    ):
+        """Create mock xarray DataArray for profile testing."""
+        if warming_levels is None:
+            warming_levels = [2.0]
+        if simulations is None:
+            simulations = ["WRF_CESM2_r1i1p1f1"]
+
+        hours_per_year = days_in_year * 24
+
+        # Create time_delta coordinate (hourly for one year)
+        time_delta_hours = np.arange(hours_per_year)
+
+        # Create data with appropriate dimensions
+        data_shape = (len(warming_levels), hours_per_year, len(simulations))
+        data = np.random.normal(20, 5, size=data_shape)  # Temperature-like data
```

Comment:
Lol this feels pretty cold. Not sure if making this warmer makes future tests more accurate, but I do see some temp values min/max comparisons, so I figured this comment was worth mentioning